unused `io_uring_context` crashing on destruction

Input:

```c++
#include <stdexec/execution.hpp>
#include <exec/linux/io_uring_context.hpp>

int main() {
  exec::io_uring_context io_uring;
}
```

Result:

```
In file included from <source>:15:
/opt/compiler-explorer/libs/stdexec/trunk/include/exec/linux/io_uring_context.hpp:42:2: warning: #warning "Your kernel is too old to support io_uring with cancellation support." [-Wcpp]
   42 | #warning "Your kernel is too old to support io_uring with cancellation support."
      |  ^~~~~~~
Execution build compiler returned: 0
Program returned: 139
output.s: /opt/compiler-explorer/libs/stdexec/trunk/include/stdexec/__detail/__intrusive_queue.hpp:47: stdexec::__queue::__intrusive_queue<_Next>::~__intrusive_queue() [with _Item = exec::__io_uring::__task; _Item* _Item::* _Next = &exec::__io_uring::__task::__next_]: Assertion `empty()' failed.
Program terminated with signal: SIGSEGV
```

--- a/include/exec/linux/io_uring_context.hpp
+++ b/include/exec/linux/io_uring_context.hpp
@@ -341,7 +341,6 @@ namespace exec {
         , __completion_queue_{__completion_queue_region_ ? __completion_queue_region_ : __submission_queue_region_, __params_}
         , __submission_queue_{__submission_queue_region_, __submission_queue_entries_, __params_}
         , __wakeup_operation_{this, __eventfd_} {
-        __wakeup_operation_.start();
       }
 
       void wakeup() {
@@ -442,6 +441,9 @@ namespace exec {
             __stop_source_.emplace();
             // Make emplacement of stop source visible to other threads and open the door for new submissions.
             __n_submissions_in_flight_.store(0, std::memory_order_release);
+          } else {
+            // This can only happen for the very first pass of run_until_stopped()
+            __wakeup_operation_.start();
           }
         }
         scope_guard __not_running{[&]() noexcept {

--- a/test/exec/test_io_uring_context.cpp
+++ b/test/exec/test_io_uring_context.cpp
@@ -58,6 +58,11 @@ class jthread {
   }
 };
 
+TEST_CASE("io_uring_context - unused context", "[types][io_uring][schedulers]") {
+  io_uring_context context;
+  CHECK(context.is_running() == false);
+}
+
 TEST_CASE("io_uring_context Satisfy concepts", "[types][io_uring][schedulers]") {
   STATIC_REQUIRE(timed_scheduler<io_uring_scheduler>);
   STATIC_REQUIRE_FALSE(std::is_move_assignable_v<io_uring_context>);

